hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java

```diff
@@ -345,7 +345,9 @@ public void onMemberListChange() {
 
     @Override
     public void onClusterVersionChange(Version newVersion) {
-        systemLogger.info("Cluster version set to " + newVersion);
+        if (!node.getVersion().asVersion().isEqualTo(newVersion)) {
```

Comment:
no need to log if the cluster version is the same as the code version. 
@vbekiaris: WDYT?

Reply:
+1

Reply:
actually I guess we want to know when a member running in compatibility mode (N-1) is switched into its code version (N). 

So perhaps we should log this event when:
1. `newVersion` differs from the code version
OR
2. the current cluster version differs from `UKNOWN` (we are upgrading)

Reply:
Unfortunately at this point, the new cluster version has already been set on `ClusterStateManager.clusterVersion` so the only way to have the previous cluster version for comparison is to maintain it in a new field in `DefaultNodeExtension` solely for handling logging.

Reply:
added here: https://github.com/hazelcast/hazelcast/pull/13743/files#diff-7d18a1f998f5c8fd7c346f0c4475ef2dR326